src/shipping/src/shipping_service/quote.rs

```diff
@@ -50,17 +49,21 @@ async fn request_quote(count: u32) -> Result<f64, Box<dyn std::error::Error>> {
     let mut reqbody = HashMap::new();
     reqbody.insert("numberOfItems", count);
 
-    let client = ClientBuilder::new(reqwest::Client::new())
-        .with(TracingMiddleware::<SpanBackendWithUrl>::new())
-        .build();
-
-    let resp = client
+    let mut response = client
         .post(quote_service_addr)
-        .json(&reqbody)
-        .send()
-        .await?
-        .text_with_charset("utf-8")
-        .await?;
+        .trace_request()
+        .send_json(&reqbody)
+        .await
+        .map_err(|err| io::Error::new(io::ErrorKind::Other, err.to_string()))?;
```

Comment:
This is weird - an application-level HTTP error doesn't really make sense wrapped into an IO error. You should either use a custom error type for this function and map back to that, or use anyhow. If you use anyhow, you can use it's `context` to add extra details - e.g.:

```
...
.send_json(...)
.await
.context("Failed to call quote service")
```